pkg/apis/networking/v1alpha1/certificate_types.go

```diff
@@ -93,4 +94,24 @@ type CertificateStatus struct {
 	// by this resource in spec.secretName.
 	// +optional
 	NotAfter *metav1.Time `json:"notAfter,omitempty"`
+
+	// HTTP01Challenges is a list of HTTP01 challenges that need to be fulfilled
+	// in order to get the TLS certificate..
+	HTTP01Challenges []HTTP01Challenge `json:"http01Challenges,omitempty"`
+}
+
+// HTTP01Challenge defines the status of a HTTP01 challenge that a certificate needs
+// to fulfill.
+type HTTP01Challenge struct {
+	// URL is the URL that the HTTP01 challenge is expected to serve on.
+	apis.URL `json:"url,omitempty"`
```

Comment:
I wouldn't inline this field, and it has to be a pointer for `omitempty` to work:
```suggestion
	URL *apis.URL `json:"url,omitempty"`
```

Reply:
Done.